easydigitaldownloads/easy-digital-downloads
# Ensure UI & CLI migrations delete the same legacy data #7824

Fixes #7824

Proposed Changes:
1. CLI - add support for deleting legacy customer emails
2. CLI - add support for deleting legacy customer addresses
3. CLI - add support for deleting legacy orders
4. UI - add support for deleting legacy customer addresses
5. UI - add support for deleting legacy notes

---

## To test:

1. Run a migration via the UI. Opt to delete legacy data at the end. Once complete, ensure all legacy data has been deleted. (see below)
2. Run a migration via WP-CLI. Opt to delete legacy data at the end. Once complete, ensure all legacy data has been deleted. (see below)

These queries can be run manually to check that all data has been deleted:

```mysql
SELECT COUNT(*) FROM wp_edd_customermeta WHERE meta_key = 'additional_email';
SELECT COUNT(*) FROM wp_usermeta WHERE meta_key = '_edd_user_address';
SELECT COUNT(*) FROM wp_posts WHERE post_type IN('edd_payment', 'edd_discount', 'edd_log');
SELECT COUNT(*) FROM wp_comments WHERE comment_type = 'edd_payment_note';
```

## Note:

In the UI, I've combined deleting post types and comments in the same process. This should work fine, but the percentage bar isn't guaranteed to be accurate. That's because we're deleting data as we go and don't keep track of exactly how many have been deleted in each step. Because although it's set up to delete a max of 25 post types and 25 comments each step, there may not necessarily be that full amount each time. You might get to a point where you're deleting 25 post types and 0 comments each step because the comments have all been fully deleted.
